From 8566a8c7185c51102f63329db1249d0cd18fff6a Mon Sep 17 00:00:00 2001
From: John Vandenberg <jayvdb@gmail.com>
Date: Wed, 3 Apr 2019 09:34:22 +0700
Subject: [PATCH] Avoid using requests unnecessarily in tests

This replaces requests with tablib in test_include_editable_packages
as in a de-vendored environment the locking fails with requests.

Continuation of https://github.com/pypa/pipenv/issues/3644
---
 news/3644.trivial.rst             | 1 +
 tests/integration/test_project.py | 8 ++++----
 2 files changed, 5 insertions(+), 4 deletions(-)
 create mode 100644 news/3644.trivial.rst

diff --git a/news/3644.trivial.rst b/news/3644.trivial.rst
new file mode 100644
index 000000000..5a7db2e18
--- /dev/null
+++ b/news/3644.trivial.rst
@@ -0,0 +1 @@
+Use tablib instead of requests in tests to avoid failures when vendored
diff --git a/tests/integration/test_project.py b/tests/integration/test_project.py
index d193fc7dc..30b4ecc53 100644
--- a/tests/integration/test_project.py
+++ b/tests/integration/test_project.py
@@ -152,16 +152,16 @@ def test_many_indexes(PipenvInstance, pypi):
 @pytest.mark.install
 @pytest.mark.project
 def test_include_editable_packages(PipenvInstance, pypi, testsroot, pathlib_tmpdir):
-    file_name = "requests-2.19.1.tar.gz"
-    package = pathlib_tmpdir.joinpath("requests-2.19.1")
-    source_path = os.path.abspath(os.path.join(testsroot, "test_artifacts", file_name))
+    file_name = "tablib-0.12.1.tar.gz"
+    package = pathlib_tmpdir.joinpath("tablib-0.12.1")
+    source_path = os.path.abspath(os.path.join(testsroot, "pypi", "tablib", file_name))
     with PipenvInstance(chdir=True, pypi=pypi) as p:
         with tarfile.open(source_path, "r:gz") as tarinfo:
             tarinfo.extractall(path=str(pathlib_tmpdir))
         c = p.pipenv('install -e {}'.format(package))
         assert c.return_code == 0
         project = Project()
-        assert "requests" in [
+        assert "tablib" in [
             package.project_name
             for package in project.environment.get_installed_packages()
         ]
